lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -284,14 +288,38 @@ sub show {
         $self->stash(ulogs       => []);
     }
 
+    my @conds;
+    push(@conds, {'me.state' => 'done'});
+    push(@conds, {'me.result' => {-not_in => [OpenQA::Schema::Result::Jobs::INCOMPLETE_RESULTS]}});
+    # instead of searching for "previous jobs" it might be better to search for same or previous builds
```

Comment:
"previous build" is unfortunately in reality very unreliable, so I think you don't really have an alternative to what you do